src/persist-client/src/cfg.rs

```diff
@@ -0,0 +1,287 @@
+            // options, where the first is always provided and the second is
+            // conditionally enabled by the process orchestrator.
+            hostname: std::env::var("HOSTNAME").unwrap_or_else(|_| "unknown".to_owned()),
+        }
+    }
+
+    /// Returns a new instance of [PersistConfig] for tests.
+    #[cfg(test)]
+    pub fn new_for_tests() -> Self {
+        use mz_build_info::DUMMY_BUILD_INFO;
+        use mz_ore::now::SYSTEM_TIME;
+
+        let mut cfg = Self::new(&DUMMY_BUILD_INFO, SYSTEM_TIME.clone());
+        cfg.hostname = "tests".into();
+        cfg
+    }
+}
+
+pub(crate) const MB: usize = 1024 * 1024;
+
+impl PersistConfig {
+    /// Default value for [`PersistConfig::blob_target_size`].
+    pub const DEFAULT_BLOB_TARGET_SIZE: usize = 128 * MB;
+    /// Default value for [`PersistConfig::compaction_minimum_timeout`].
+    pub const DEFAULT_COMPACTION_MINIMUM_TIMEOUT: Duration = Duration::from_secs(90);
+
+    // Move this to a PersistConfig field when we actually have read leases.
+    //
+    // MIGRATION: Remove this once we remove the ReaderState <->
+    // ProtoReaderState migration.
+    pub(crate) const DEFAULT_READ_LEASE_DURATION: Duration = Duration::from_secs(60 * 15);
+
+    // Tuning notes: Picked arbitrarily.
+    pub(crate) const NEED_ROLLUP_THRESHOLD: u64 = 128;
+}
+
+impl ConsensusKnobs for PersistConfig {
+    fn connection_pool_max_size(&self) -> usize {
+        self.consensus_connection_pool_max_size
+    }
+
+    fn connection_pool_ttl(&self) -> Duration {
+        self.consensus_connection_pool_ttl
+    }
+
+    fn connection_pool_ttl_stagger(&self) -> Duration {
+        self.consensus_connection_pool_ttl_stagger
+    }
+}
+
+/// Updates to values in [PersistConfig].
+///
+/// These reflect updates made to LaunchDarkly. They're passed from environmentd
+/// through storage and compute commands and applied to PersistConfig to change
+/// its values.
+///
+/// Parameters can be set (`Some`) or unset (`None`). Unset parameters should be
+/// interpreted to mean "use the previous value".
+#[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize, Arbitrary)]
+pub struct PersistParameters {
+    /// Configures [`PersistConfig::blob_target_size`].
+    pub blob_target_size: Option<usize>,
+    /// Configures [`PersistConfig::compaction_minimum_timeout`].
+    pub compaction_minimum_timeout: Option<Duration>,
+}
+
+impl PersistParameters {
+    /// Update the parameter values with the set ones from `other`.
+    pub fn update(&mut self, other: PersistParameters) {
+        if let Some(v) = other.blob_target_size {
+            self.blob_target_size = Some(v);
+        }
+        if let Some(v) = other.compaction_minimum_timeout {
+            self.compaction_minimum_timeout = Some(v);
+        }
+    }
+
+    /// Return whether all parameters are unset.
+    pub fn all_unset(&self) -> bool {
+        self.blob_target_size.is_none() && self.compaction_minimum_timeout.is_none()
```

Comment:
where is this method used / could we do the destructuring trick to make sure we always consider every field?